[camera] Add web support

bselwe opened the pull request:
Adds web support to the camera plugin.

- Added web support to the camera `example` app, following the [instructions](https://github.com/flutter/plugins/blob/master/packages/camera/camera_web/README.md) of the web package.
- Updated camera README.md with the web instructions.

### TODO
- [x] Endorse `camera_web` plugin.
- [x] Update `camera` version (pubspec.yaml) and CHANGELOG.md. 

Part of https://github.com/flutter/flutter/issues/45297.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

bselwe commented:
@ditman @felangel

bselwe commented on packages/camera/camera/pubspec.yaml:
```diff
@@ -18,9 +18,12 @@ flutter:
 
 dependencies:
   camera_platform_interface: ^2.0.0
+  camera_web: ^0.0.1
```
This has to be updated with the released version.

ditman commented:
I'm going to use this to publish a demo version of the camera_web

ditman commented:
**Published here!**

https://dit-vgv-camera-test.web.app

There's something weird going on with the orientation of the preview in my machine. It seems to be preferring a portrait camera rather than landscape in my laptop. Also in my phone the external camera is flipped, when it probably shouldn't? Not sure, but it's a start!! :D

sgehrman commented:
> There's something weird going on with the orientation of the preview in my machine. It seems to be preferring a portrait camera rather than landscape in my laptop

Same issue.  Works great but I can't figure out how to force it to landscape on web.

ditman commented:
OK, so I've deployed the latest changes to https://dit-vgv-camera-test.web.app, and now there's only one thing that bugs me slightly: the preview does not really give users a clue of what's the orientation of the camera when the photo is taken. This is especially true in mobile phones (where the preview is landscape, but the photo taken is normally in portrait mode).

However, I think this is an issue with the example app itself, rather than the plugin itself. For now, I don't think this is a release blocker, so we probably should publish the first version of the camera plugin asap, and get it in the hands of the people :) (well, more people, at least!)

bselwe commented:
@ditman Hm, not sure but it seems that the published version did not include some of the latest PRs/fixes. Rebased with master and reverted one unnecessary fix.

ditman commented:
@bselwe I've deployed the latest demo, and everything that has been merged so far, here:

https://dit-vgv-camera-test.web.app

(Verified it contains up to "don't flip the preview on external cameras")

ditman commented:
This needs to be rebased one last time after https://github.com/flutter/plugins/pull/4358 lands, so we only merge code changes in the `camera` core package.

ditman commented:
This looks good, but I'm not endorsing a plugin at 7:30PM of a Friday, next week :P

ditman reviewed (approved):
LGTM, let's go!
